### PR TITLE
CP-31842: implement xenstore-chmod for golang guest xenstore

### PR DESCRIPTION
**Pull request by TalonsLee**
Signed-off-by: Talons Lee <xin.li@citrix.com>

### CODE REVIEW

File: xenstore/xenstore.go [hunk 1]
```diff
@@ -222,6 +225,53 @@ func xs_ls(script_name string, args []string) {
 	}
 }
 
+func xs_chmod(script_name string, args []string) {
+	if len(args) == 0 || args[0] == "-h" || len(args) < 2 {
```
Comment (liulinC) [inline xenstore/xenstore.go]: `if len(args) < 2 || args[0] == "-h" `?

File: xenstore/xenstore.go [hunk 2]
```diff
@@ -222,6 +225,53 @@ func xs_ls(script_name string, args []string) {
...
+			case 'w':
+				p.Pe = xenstoreclient.PERM_WRITE
+			case 'b':
+				p.Pe = xenstoreclient.PERM_READWRITE
+			default:
+				err = errors.New("Invalid mode string")
+			}
+			if err == nil {
+				var id uint64
+				id, err = strconv.ParseUint(m[1:], 10, 0)
```
Comment (liulinC) [inline xenstore/xenstore.go]: what is the magic number `10`? is it possible to assign the value to  a meaningful name?

Comment (TalonsLee) [reply xenstore/xenstore.go]: 10 is the base for conversion.
it's OK to use it in this way:
see this example in std lib:
https://github.com/golang/go/blob/master/src/strconv/atoi.go#L247

File: xenstore/xenstore.go (hunk 1 above)
Comment (TalonsLee) [reply xenstore/xenstore.go]: Done

File: xenstore/xenstore.go [hunk 3]
```diff
@@ -222,6 +225,53 @@ func xs_ls(script_name string, args []string) {
...
+				if err == nil {
+					p.Id = uint(id)
+					perms = append(perms, p)
+				}
+			}
+		} else {
+			err = errors.New("Invalid mode string")
```
Comment (minli1) [inline xenstore/xenstore.go]: Here the error message `Invalid mode string` is same as the Line 250, when this kind of error appear then we can't determine where is the exact place. Suggest giving a different error message.

Comment (TalonsLee) [reply xenstore/xenstore.go]: OK. Set a different err message.

**Pull request merged**